### ISSUE TITLE
azurerm_cognitive_account services specified in Kind validation function need updated

### ISSUE THREAD

**Issue by neilbailie** [terraform-providers/terraform-provider-azurerm#2294]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform v0.11.8
+ provider.azurerm v1.18.0

### Affected Resource(s)

* azurerm_cognitive_account

I think Microsoft has combined a number of Speech services under a new Kind which can't be consumed in Terraform as the function to validate Kind doesn't include the new type.

At the same time if I try to consume one of the older Kind services Azure complains that the service isn't available.

I can consume a Kind of LUIS without issue as this is still available in Azure.

### Terraform Configuration Files

```hcl

resource "azurerm_cognitive_account" "cognitive_speech_service" {
  name                = "speech001"
  location            = "eastus2"
  resource_group_name = "test2"
  kind                = "SpeechServices"

  sku {
    name = "S0"
    tier = "Standard"
  }
}

```
Or

```hcl

resource "azurerm_cognitive_account" "cognitive_speech_service" {
  name                = "speech001"
  location            = "eastus2"
  resource_group_name = "test2"
  kind                = "Bing.Speech"

  sku {
    name = "S0"
    tier = "Standard"
  }
}

```
### Expected Behavior

SpeechServices API service created

### Actual Behavior

* Error: module.cognitive_services_speech_service.azurerm_cognitive_account.cognitive_speech_service: expected kind to be one of [Academic Bing.Autosuggest Bing.Autosuggest.v7 Bing.CustomSearc
h Bing.Search Bing.Search.v7 Bing.Speech Bing.SpellCheck Bing.SpellCheck.v7 ComputerVision ContentModerator CustomSpeech Emotion Face LUIS Recommendations SpeakerRecognition Speech SpeechTranslation TextAnalytics TextTranslation WebLM], got SpeechServices

OR

* azurerm_cognitive_account.cognitive_speech_service: Error creating Cognitive Services Account "test" (Resource Group "test"): cognitiv
eservices.AccountsClient#Create: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="InvalidApiSetId" Message="The ac
count type 'Bing.Speech' is either invalid or unavailable in given region."

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by neilbailie** [comment]
I'd be interested in working on this issue.

**Comment by tombuildsstuff** [comment]
hey @neilbailie 

Thanks for opening this issue :)

Taking a quick look into this [it appears this value is missing](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/vendor/github.com/Azure/azure-sdk-for-go/services/cognitiveservices/mgmt/2017-04-18/cognitiveservices/models.go#L45-L84) from the possible values for the 'kind' field.

Whilst we can look to add this value to [the validation function](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_cognitive_account.go#L42-L65) - I have a feeling it might be worth [opening an issue/PR on the Rest API Specs Repository](https://github.com/Azure/azure-rest-api-specs) to get these added to [the Swagger file](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2017-04-18/cognitiveservices.json#L718-L738) which the SDK is generated from.

Whilst we could add this to the validation function outlined above, it'd be parsed incorrectly since it's not part of the Enum/Constant outlined above - as such I believe adding this to the Swagger (and thus the Go SDK) would be the best place to start here.

Hope that helps :)

Thanks!

**Comment by mb290** [comment]
@neilbailie we were having the same issue with Bing.Search.v7 and the same error about location. Upon reverse engineering the ARM Templates we actually found for location we needed to specify "global" and not "eastus2" as it's considered a global service from MS. It applied in TF and appeared in the portal so maybe worth a go?

**Comment by tomrijntjes** [comment]
@mb290 changing to "global" doesn't help. Opening a cognitive service in the portal clearly lists "West Europe", not "Global". I'm not sure how you found that MS considers it a global service but it's definitely not reflected in the UI.

> Status=400 Code="InvalidApiSetId" Message="The account type 'Speech' is either invalid or unavailable in given region.

Other suggestions for workarounds?

**Comment by neilbailie** [comment]
Hi @tombuildsstuff 

I request Microsoft make the changes suggested by you and I believe the Swagger and Go SDK (version 24.0.0) have now been updated to use a string instead of a enum for the kind and SKU.

**Comment by katbyte** [comment]
@neilbailie,

Thank you for the update, I can confirm that `v24` of the SDK has changed to a string. I have a [open PR](https://github.com/terraform-providers/terraform-provider-azurerm/pull/2572) for the upgrade but it is blocked on another bug in the SDK so this is currently blocked.

We will still need to find a list of all possible values (probably from the portal) so we can continue doing validation on plan.

**Comment by neilbailie** [comment]
Now works as expected under the v24 SDK.
